src/session/user.rs

```diff
@@ -276,14 +316,49 @@ impl User {
             gtk::PropertyExpression::new(User::static_type(), Some(user_expression), "first-name");
         let last_name_expression =
             gtk::PropertyExpression::new(User::static_type(), Some(user_expression), "last-name");
+        let type_expression =
+            gtk::PropertyExpression::new(User::static_type(), Some(user_expression), "type");
         gtk::ClosureExpression::with_callback(
-            &[first_name_expression, last_name_expression],
+            &[first_name_expression, last_name_expression, type_expression],
             |args| {
+                let type_ = args[3].get::<BoxedUserType>().unwrap().0;
                 let first_name = args[1].get::<String>().unwrap();
                 let last_name = args[2].get::<String>().unwrap();
-                format!("{} {}", first_name, last_name).trim().to_owned()
+                if type_ == UserType::Deleted {
+                    gettext("Deleted Account")
+                } else {
+                    format!("{} {}", first_name, last_name).trim().to_owned()
+                }
             },
         )
         .upcast()
     }
+
+    pub fn formatted_status_expression(user_expression: &gtk::Expression) -> gtk::Expression {
+        let type_expression =
+            gtk::PropertyExpression::new(User::static_type(), Some(user_expression), "type");
+        let status_expression =
+            gtk::PropertyExpression::new(User::static_type(), Some(user_expression), "status");
+        gtk::ClosureExpression::with_callback(&[status_expression, type_expression], |args| {
+            let user_status = &args[1].get::<BoxedUserStatus>().unwrap().0;
+            let type_ = &args[2].get::<BoxedUserType>().unwrap().0;
+            match type_ {
+                UserType::Regular => match user_status {
+                    UserStatus::Online(_) => gettext("online"),
+                    UserStatus::Offline(data) => {
+                        let unix = data.was_online;
+                        super::user::last_seen_time_ago(unix)
+                    }
+                    UserStatus::Recently => gettext("last seen recently"),
+                    UserStatus::LastWeek => gettext("last seen last week"),
+                    UserStatus::LastMonth => gettext("last seen last month"),
```

Comment:
Actually, these should be "last seen within a week" and "last seen within a month" to be consistent with official clients.